var/spack/repos/builtin/packages/perl/package.py

```diff
@@ -63,16 +64,29 @@ class Perl(Package):  # Perl doesn't use Autotools, it should subclass Package
 
     extendable = True
 
-    depends_on('gdbm')
+    # :5.28 needs gdbm@:1:14.1: https://rt-archive.perl.org/perl5/Ticket/Display.html?id=133295
+    depends_on('gdbm',         when='@5.30.0:')
+    depends_on('gdbm@:1.14.1', when='@:5.28.0')
     depends_on('berkeley-db')
     depends_on('bzip2+shared')
-    depends_on('zlib+shared')
+    # :5.24.1 needs zlib@:1.2.8: https://rt.cpan.org/Public/Bug/Display.html?id=120134
+    depends_on('zlib+shared',        when='@5.26.2:')
+    depends_on('zlib+shared@:1.2.8', when='@5.20.3:5.24.1')
```

Comment:
The `+shared` constraint is orthogonal to the version constraint, so we can write those like this:
```suggestion
    depends_on('zlib+shared')
    # :5.24.1 needs zlib@:1.2.8: https://rt.cpan.org/Public/Bug/Display.html?id=120134
    depends_on('zlib@:1.2.8', when='@5.20.3:5.24.1')
```